pkg/github/github_manager.go

```diff
@@ -0,0 +1,99 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/ghodss/yaml"
+	klum "github.com/jadolg/klum/pkg/apis/klum.cattle.io/v1alpha1"
+	log "github.com/sirupsen/logrus"
+)
+
+func UploadGithubSecret(userSync *klum.UserSync, kubeconfig *klum.Kubeconfig, githubURL string, githubToken string) error {
+	githubSync := userSync.Spec.Github
+	if !githubSync.Valid() {
+		return fmt.Errorf("not enough github data to be able to create a GitHub secret")
+	}
+
+	kubeconfigYAML, err := toYAMLString(kubeconfig.Spec)
+	if err != nil {
+		return err
+	}
+
+	kubeconfigYAMLb64 := base64.StdEncoding.EncodeToString([]byte(kubeconfigYAML))
+
+	latestKubeconfigUploaded, present := userSync.Annotations["klum.cattle.io/lastest.upload.github"]
+	if present && latestKubeconfigUploaded == kubeconfigYAMLb64 {
+		return nil
+	}
+	userSync.Annotations["klum.cattle.io/lastest.upload.github"] = kubeconfigYAMLb64
```

Comment:
I'll use a hash instead of b64. I thought might be useful to be able to check the contents for debugging, but we might as well just hash the other side and compare.

Reply:
Changed this in [032823d](https://github.com/jadolg/klum/pull/19/commits/032823da5bbadd97a5a8af908830fea136974bf6)